src/workerd/api/filesystem.c++

```diff
@@ -273,39 +295,51 @@ void FileSystemModule::link(jsg::Lock& js, FilePath from, FilePath to, LinkOptio
   const jsg::Url& fromUrl = normalizedFrom;
   const jsg::Url& toUrl = normalizedTo;
 
-  JSG_REQUIRE(vfs.resolve(js, fromUrl) == kj::none, Error, "File already exists"_kj);
+  if (vfs.resolve(js, fromUrl) != kj::none) {
+    throwUVException(js, UV_EEXIST, "link"_kj, "File already exists"_kj);
+  }
+
   // Now, let's split the fromUrl into a base directory URL and a file name so
   // that we can make sure the destination directory exists.
   jsg::Url::Relative fromRelative = fromUrl.getRelative();
-  JSG_REQUIRE(fromRelative.name.size() > 0, Error, "Invalid filename"_kj);
 
-  auto parent =
-      JSG_REQUIRE_NONNULL(vfs.resolve(js, fromRelative.base), Error, "Directory does not exist"_kj);
-  auto& dir = JSG_REQUIRE_NONNULL(
-      parent.tryGet<kj::Rc<workerd::Directory>>(), Error, "Invalid argument"_kj);
+  if (fromRelative.name.size() == 0) {
+    throwUVException(js, UV_EINVAL, "linn"_kj, "Invalid filename"_kj);
```

Comment:
Windsurf spotted this typo but it totally garbled its suggestion, so let me do it right:

```suggestion
    throwUVException(js, UV_EINVAL, "link"_kj, "Invalid filename"_kj);
```